include/micm/solver/lu_decomposition.hpp

```diff
@@ -108,16 +108,6 @@ namespace micm
         const SparseMatrixPolicy& A,
         typename SparseMatrixPolicy::value_type initial_value);
 
-    /// @brief Perform an LU decomposition on a given A matrix
-    /// @param A Sparse matrix to decompose
-    /// @param L The lower triangular matrix created by decomposition
-    /// @param U The upper triangular matrix created by decomposition
-    template<class SparseMatrixPolicy>
-    requires(SparseMatrixConcept<SparseMatrixPolicy>) void Decompose(
-        const SparseMatrixPolicy& A,
-        SparseMatrixPolicy& L,
-        SparseMatrixPolicy& U) const;
```

Comment:
What is it used for before? I don't think we ever use this function?

Reply:
It was a pass through. When we called this, we declared a local boolean variable and then called the actual function with the boolean variable. Since this is on our hot path, I figured it's a slight, and useful, optimization to remove one branch (in the LinearFactor function of rosenbrock) and one stack call by eliminating this pass through

Reply:
Thanks for the clarification. That makes sense.